packages/cell-toolbar/src/celltoolbartracker.ts

```diff
@@ -322,13 +365,22 @@ export class CellToolbarTracker implements IDisposable {
       return false; // Nothing in the editor
     }
 
-    const codeMirrorLines =
-      editorWidget.node.getElementsByClassName('CodeMirror-line');
+    const codeMirrorLines = editorWidget.node.getElementsByClassName('cm-line');
     if (codeMirrorLines.length < 1) {
       return false; // No lines present
     }
-    const lineRight = codeMirrorLines[0].children[0] // First span under first pre
-      .getBoundingClientRect().right;
+
+    let lineRight = codeMirrorLines[0].getBoundingClientRect().left;
+    const range = document.createRange();
+    range.setStart(codeMirrorLines[0].childNodes[0], 0);
+    const nodeLength = codeMirrorLines[0].childNodes.length;
+    const lastNode = codeMirrorLines[0].childNodes[nodeLength - 1];
+    range.setEnd(
+      lastNode,
+      Math.max((lastNode?.textContent?.length || 0) - 1, 0)
+    );
```

Comment:
Can we try?

```suggestion
    range.selectNodeContents(codeMirrorLines[0]);
```

Reply:
Since the other suggestion was a miss, what do you think about this one?

Reply:
This works! Thanks @krassowski